shim/shim.py

```diff
@@ -755,6 +755,154 @@ def _get_command_options(
+
+    if arguments.shims_paths:
+        options_json["shims-paths"] = arguments.shims_paths
+
+    if arguments.graphql_metadata_paths:
+        options_json["graphql-metadata-paths"] = arguments.graphql_metadata_paths
+
+    if arguments.source_exclude_directories:
+        options_json["source-exclude-directories"] = arguments.source_exclude_directories
+
+    if arguments.generated_models_directory:
+        options_json["generated-models-directory"] = arguments.generated_models_directory
+
+    if arguments.sharded_models_directory:
+        options_json["sharded-models-directory"] = arguments.sharded_models_directory
+
+    if arguments.emit_all_via_cast_features:
+        options_json["emit-all-via-cast-features"] = arguments.emit_all_via_cast_features
+
+    if arguments.propagate_across_arguments:
+        options_json["propagate-across-arguments"] = arguments.propagate_across_arguments
+
+    if arguments.allow_via_cast_feature:
+        options_json["allow-via-cast-feature"] = []
+        for feature in arguments.allow_via_cast_feature:
+            options_json["allow-via-cast-feature"].append(feature.strip())
+
+    if arguments.heuristics:
+        options_json["heuristics"] = arguments.heuristics
+
+    if arguments.sequential:
+        options_json["sequential"] = arguments.sequential
+
+    if arguments.skip_source_indexing:
+        options_json["skip-source-indexing"] = arguments.skip_source_indexing
+
+    if arguments.skip_analysis:
+        options_json["skip-analysis"] = arguments.skip_analysis
+
+    if arguments.disable_parameter_type_overrides:
+        options_json["disable-parameter-type-overrides"] = arguments.disable_parameter_type_overrides
+
+    if arguments.disable_global_type_analysis:
+        options_json["disable-global-type-analysis"] = arguments.disable_global_type_analysis
+
+    if arguments.verify_expected_output:
+        options_json["verify-expected-output"] = arguments.verify_expected_output
+
+    if arguments.remove_unreachable_code:
+        options_json["remove-unreachable-code"] = arguments.remove_unreachable_code
+
+    if arguments.maximum_method_analysis_time is not None:
+        options_json["maximum-method-analysis-time"] = arguments.maximum_method_analysis_time
+
+    if arguments.enable_cross_component_analysis:
+        options_json["enable-cross-component-analysis"] = arguments.enable_cross_component_analysis
+
+    if arguments.extra_analysis_arguments:
+        for arg in shlex.split(arguments.extra_analysis_arguments):
+            options_json[arg] = arg
+
+    if arguments.job_id:
+        options_json["job-id"] = arguments.job_id
+
+    if arguments.metarun_id:
+        options_json["metarun-id"] = arguments.metarun_id
+
+    trace_settings = [f"MARIANA_TRENCH:{arguments.verbosity}"]
+    if "TRACE" in os.environ:
+        trace_settings.insert(0, os.environ["TRACE"])
+    os.environ["TRACE"] = ",".join(trace_settings)
```

Comment:
This is not really related to the options file, can we move this in a different function, for instance `set_environment_variables(arguments)` or something?